### PR TITLE
💄👽️(UI+API) Report Button for users

### PR DESCRIPTION
**Pull request by ricecakesreece**
Added report buttons, changed API names to better suit all users, added testing for multiple report related pages.

### CODE REVIEW

File: libs/au-pair-dashboard/feature/src/lib/user-report-modal/user-report-modal.component.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  async reportUser(formData : any){ 
+    if(formData.desc == "") {
+      this.openToast("Please add a description");
+      return;
+    }
+    
+    this.sending = true;
+
```
Comment (Ruben-Brits) [inline libs/au-pair-dashboard/feature/src/lib/user-report-modal/user-report-modal.component.ts:53]: empty line

File: libs/parent-dashboard/feature/src/lib/parent-dashboard.component.spec.ts [hunk 2]
```diff
@@ -50,9 +64,133 @@ describe('ParentProfileComponent', () => {
...
+    component.openToast("test");
+    expect(await component.openToast).toReturn();
+  });
+
+  it('should,close the modal when closeModal is called', async ()=>{
+    jest.spyOn(component,"closeModal");
+    component.closeModal();
+    expect(await component.closeModal).toReturn();
+  });
+});
```
Comment (Ruben-Brits) [inline libs/parent-dashboard/feature/src/lib/parent-dashboard.component.spec.ts:196]: newline

File: libs/parent-dashboard/feature/src/lib/user-report-modal/user-report-modal.component.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  async reportUser(formData : any){ 
+    if(formData.desc == "") {
+      this.openToast("Please add a description");
+      return;
+    }
+    
+    this.sending = true;
+
```
Comment (Ruben-Brits) [inline libs/parent-dashboard/feature/src/lib/user-report-modal/user-report-modal.component.ts:53]: empty line

**Review by Ruben-Brits — CHANGES_REQUESTED** [review]

**Review by PxttytheThxtty — DISMISSED** [review]

**Review by Ruben-Brits — DISMISSED** [review]

**Comment by Ruben-Brits** [comment]
Please sort out merge conflict

**Review by priyo5 — APPROVED** [review]

**Review by priyo5 — APPROVED** [review]

**Review by Ruben-Brits — APPROVED** [review]

**Pull request merged**